client/components/wcc-settings-form/index.js

```diff
@@ -9,19 +9,76 @@ import { successNotice, errorNotice } from 'state/notices/actions';
 import * as SettingsActions from 'state/settings/actions';
 import * as PackagesActions from 'state/form/packages/actions';
 import { getFormErrors } from 'state/selectors';
+import { translate as __ } from 'lib/mixins/i18n';
+import FormButton from 'components/forms/form-button';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
 
 const WCCSettingsForm = ( props ) => {
+	const currentStep = props.form.currentStep;
+	const currentStepLayout = props.layout[ currentStep ];
+
+	const renderCurrentStep = () => {
+		if ( ! currentStepLayout ) {
+			return <span>... spinner or something ...</span>;
```

Comment:
So here's the thing: The form starts in the step `-1`. During this time, the form should display an spinner or something.
Right when the user lands in a step, `something` must be triggered so that the form will auto-advance to the next field, and so on, if the current step's form is in a correct state (according to the client **and** the server). Since this auto-advance needs server-side validation, some kind of spinner is needed here.

Of course, none of this is implemented yet, so you must manually click the "Next" button to get to the first form.